examples/binaries/constructor/src/builder.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{Arg, Call};
+use alloc::{boxed::Box, string::ToString, vec::Vec};
+use core::{fmt::Debug, ops::Deref};
+use parity_scale_codec::{WrapperTypeDecode, WrapperTypeEncode};
+
+#[derive(Default, Debug, Clone)]
+pub struct Calls(Vec<Call>);
+
+impl From<Vec<Call>> for Calls {
+    fn from(calls: Vec<Call>) -> Self {
+        Self(calls)
+    }
+}
+
+impl Deref for Calls {
```

Comment:
Maybe it is worth to impl `DerefMut` too, to use some `Vec`'s methods?

Reply:
Implemented deref only for encode_like trait bound. I don't want inner access here at all, but deref also allows to call for example .len, so its okay without mutable impl for me